### ISSUE TITLE
Popeye CHOCKED! "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by hemantdindi** [derailed/popeye#148]
<img src="https://raw.githubusercontent.com/derailed/popeye/master/assets/popeye_bug.png" align="right" width="100" height="auto"/>

<br/>
<br/>
<br/>

**Describe the bug**
When running popeye as a non-root image, the process is in hung state and we see the below msg in the logs -

```
6:25AM ERR Popeye CHOCKED! "invalid memory address or nil pointer dereference"
6:25AM ERR goroutine 200 [running]:
runtime/debug.Stack(0x255d480, 0x1919003, 0x0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/derailed/popeye/pkg.(*Popeye).sanitizer.func1()
        /popeye/pkg/popeye.go:290 +0xfa
panic(0x172d380, 0x2540e00)
        /usr/local/go/src/runtime/panic.go:969 +0x175
github.com/derailed/popeye/internal/sanitize.(*Pod).checkOwnedByAnything(0xc0007323e0, 0x1b69140, 0xc000581260, 0xc0007beaa0, 0x1, 0x1)
        /popeye/internal/sanitize/pod.go:116 +0x9c
github.com/derailed/popeye/internal/sanitize.(*Pod).Sanitize(0xc0007323e0, 0x1b69140, 0xc000581140, 0xc00046c210, 0xc0006b4f2a)
        /popeye/internal/sanitize/pod.go:82 +0x2fc
github.com/derailed/popeye/internal/scrub.(*Pod).Sanitize(0xc00046c210, 0x1b69140, 0xc0003be990, 0xc0000100e8, 0x1b565c0)
        /popeye/internal/scrub/pod.go:46 +0x8e
github.com/derailed/popeye/pkg.(*Popeye).sanitizer(0xc00008d130, 0x1b69140, 0xc0003be990, 0x190c501, 0x7, 0x0, 0x0, 0x190c501, 0x2, 0x190c504, ...)
        /popeye/pkg/popeye.go:295 +0xcf
created by github.com/derailed/popeye/pkg.(*Popeye).sanitize
        /popeye/pkg/popeye.go:265 +0x865
```

**To Reproduce**
1. Build a new non-root image and run on the k8s cluster-

```
FROM quay.io/derailed/popeye
RUN addgroup -g 1001 popeye && adduser -u 1001 -G popeye -h /home/popeye -D popeye
RUN chown popeye:popeye /bin/popeye && chmod +x /bin/popeye
USER popeye
```

**Expected behavior**
The results should be displayed without any issue

**Screenshots**
NA

**Versions (please complete the following information):**
 - OS: [e.g. OSX]
 - Popeye - v0.9.0
 - K8s - 1.19

**Comment by derailed** [comment]
@hemantdindi Thank you, great catch! I push a fix in the next drop!

**Comment by hemantdindi** [comment]
Getting results when using individual components
```
popeye -s svc  -s cm  -s pv  -s pvc  -s secrets  -s deploy  -s statefulsets  -s horizontalpodautoscalers  -s namespace  -s role  -s rolebinding  -s ingress -s sa -s daemonsets -s replicasets -s node  -s clusterrole -s clusterrolebinding -s networkpolicy -s podsecuritypolicy
```
Below command fails with the `choked' msg
```
popeye -s pod
```

**Comment by derailed** [comment]
@hemantdindi Fixed v0.9.1
